fastestimator/estimator/trace.py

```diff
@@ -195,6 +195,95 @@ def on_epoch_end(self, state):
 
     def on_end(self, state):
         state['total_time'] = "{} sec".format(round(time.perf_counter() - self.train_start, 2))
+        for model_name, model in self.network.model.items():
+            state[model_name + "_lr"] = round(backend.get_value(model.optimizer.lr), 6)
+
+
+class LRController(Trace):
+    def __init__(self,
+                 model,
+                 lr_schedule=None,
+                 reduce_on_eval=False,
+                 reduce_patience=10,
+                 reduce_factor=0.1,
+                 reduce_metric="loss",
```

Comment:
why separate reduce_on_eval from reduce_metric? we can just infer reduce_on_eval from whether a non-None reduce_metric is provided

Reply:
fixed.